lib/plugin.js

```diff
@@ -69,7 +69,47 @@ module.exports = class Plugin {
         return plugin;
     }
 
-    isCompatibleWithShopwareVersion(shopwareVersion) {
-        return semver.satisfies(shopwareVersion, this.shopwareCompatibility);
+    isCompatibleWithShopwareVersion(shopwareMarketingVersion) {
+        if (!shopwareMarketingVersion) {
+            return false;
+        }
+
+        if (this.shopwareMajorVersion === 6) {
+            if (!shopwareMarketingVersion.startsWith('6.')) {
+                return false;
+            }
+
+            const shopwareVersion = this.getShopware6Semver(shopwareMarketingVersion);
+            const pluginShopwareCompatibility = this.getShopware6Semver(this.shopwareCompatibility);
+
+            return semver.satisfies(shopwareVersion, pluginShopwareCompatibility);
+        }
+
+        if (this.shopwareMajorVersion === 5) {
+            if (!shopwareMarketingVersion.startsWith('5.')) {
+                return false;
+            }
+
+            return semver.satisfies(shopwareMarketingVersion, this.shopwareCompatibility);
+        }
+
+        throw new Error(
+            'The scs-commander is incompatible with the given Shopware version number'
+                + ` (${shopwareMarketingVersion}).`,
+        );
+    }
+
+    getShopware6Semver(version) {
+        if (!version.startsWith('6.')) {
+            throw new Error('No Shopware 6 version number given.');
+        }
+
+        const versionPieces = version.split('.').length + 1;
```

Comment:
Why do we have to add one to the length here? My understanding from reading the code following is that passing a value of `6.1.0` to this function results in `versionPieces = 4` and a return value of `1.0` while passing `6.1.0.1` results in an error. Is this the desired outcome?

On another note: `versionPieces` does not contain any pieces, but the number of pieces/parts/components:
```suggestion
        const numberOfVersionComponents = version.split('.').length + 1;
```